gulpfile.js

```diff
@@ -6,11 +6,35 @@
   its own file in `/gulp`. Any files in that directory
   get automatically required below.
   To add a new task, simply add a new task file that directory.
-  `/gulp/tasks.js` specifies the default set of
-  tasks to run when you run `gulp`.
 */
 
+const gulp = require('gulp')
 const requireDir = require('require-dir')
 
 // Require all tasks in gulp/tasks, including subfolders
-requireDir('./gulp', {recurse: true})
+requireDir('./gulp', { recurse: true })
+
+// gulp 4 requires dependency tasks to be defined before they are called.
+// We'll keep our top-level tasks in this file so that they are defined at the end of the chain, after their dependencies.
+gulp.task('generate-assets', gulp.parallel(
+  'clean',
```

Comment:
I think we probably want to run `clean` before the others, not in parallel - or there's a chance `clean` could clean out files generated by the other tasks. Previously this list was series (runSequence) but maybe `clean` should be series, then the others as parallel?

Reply:
For reference here's the original file: https://github.com/alphagov/govuk-prototype-kit/blob/master/gulp/tasks.js